proxy/http/HttpBodyFactory.cc

```diff
@@ -372,7 +352,7 @@ HttpBodyFactory::HttpBodyFactory()
 HttpBodyFactory::~HttpBodyFactory()
 {
   // FIX: need to implement destructor
-  delete table_of_sets;
+  table_of_sets.reset(nullptr);
```

Comment:
release only releases the ownership and won't call the destructor as I remembered. It may cause a memory leak: http://www.cplusplus.com/reference/memory/unique_ptr/release/

Reply:
Yes - the entire point of `release` is to prevent calling the destructor on the owned object.

Reply:
@chitianhao The example from std::unique_ptr::release is more explanatory:
```
  manual_pointer = auto_pointer.release();
  delete manual_pointer;
```
vs
`table_of_sets.reset(nullptr);`

Reply:
A `.reset()` would suffice and you don't need a parameter.

Reply:
Yes, that's right. But maybe the way of `.reset(nullptr)` is clearer meaning we want to free it. `.reset()` is also fine.